### PR TITLE
Definitions for siege

### PR DESCRIPTION
**Pull request by Goldsmith42**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/siege/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    with304: Siege;
+    without304?: Siege;
+    concurrent(n: number): Siege;
+    for(n: number): For<Siege>;
+    get(url: string, query?: Stringifiable): Siege;
+    post(url: string, params: Stringifiable): Siege;
+    attack(): void;
+}
+
+declare function Siege(options?: Options): Siege;
```
Comment (peterblazejewicz) [inline types/siege/index.d.ts:85]:
```suggestion
declare function siege(options?: Options): Siege;
```
module just uses anonymous function, we can just make it easier here to distinguish what is exported
function and what is the 'Siege' class IMO

File: types/siege/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+declare function Siege(path?: string, options?: Options): Siege;
```
Comment (peterblazejewicz) [inline types/siege/index.d.ts:86]:
```suggestion
declare function siege(path?: string, options?: Options): Siege;
```

File: types/siege/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+export = Siege;
```
Comment (peterblazejewicz) [inline types/siege/index.d.ts:88]:
```suggestion
export = siege;
```

**Comment by peterblazejewicz** [comment]
I'd propbably mark properties like `withCookie` a readonly only,
note those are getters, one cannot write to them, while with current implementation,
these are just properties (write/read)

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Goldsmith42 thx!

**Comment by orta** [comment]
Looks good here 👍🏻

**Pull request merged**